dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -147,77 +152,86 @@ public String telnet(Channel channel, String message) {
             for (int ii = 0; ii < list.size(); ii++) {
                 if (list.get(ii) instanceof JSONObject) {
                     JSONObject jsonObject = (JSONObject) list.get(ii);
-                    listOfActualClass.add(jsonObject.toJavaObject(paramClasses[ii]));
+                    listOfActualClass.add(jsonObject.toJavaObject(paramTypes[ii]));
                 } else {
                     listOfActualClass.add(list.get(ii));
                 }
             }
             list = listOfActualClass;
         }
-        Invoker<?> invoker = null;
+
         Method invokeMethod = null;
-        for (Exporter<?> exporter : DubboProtocol.getDubboProtocol().getExporters()) {
-            if (service == null || service.length() == 0) {
-                invokeMethod = findMethod(exporter, method, list, paramClasses);
-                if (invokeMethod != null) {
-                    invoker = exporter.getInvoker();
-                    break;
-                }
-            } else {
-                if (service.equals(exporter.getInvoker().getInterface().getSimpleName())
-                        || service.equals(exporter.getInvoker().getInterface().getName())
-                        || service.equals(exporter.getInvoker().getUrl().getPath())) {
-                    invokeMethod = findMethod(exporter, method, list, paramClasses);
-                    invoker = exporter.getInvoker();
-                    break;
-                }
+        ProviderModel selectedProvider = null;
+        for (ProviderModel provider : ApplicationModel.allProviderModels()) {
+            if (isServiceMatch(service, provider)) {
+                invokeMethod = findMethod(provider.getAllMethods(), method, list, paramTypes);
+                selectedProvider = provider;
+                break;
             }
         }
-        if (invoker != null) {
+
+        if (selectedProvider != null) {
             if (invokeMethod != null) {
                 try {
-                    Object[] array = PojoUtils.realize(list.toArray(), invokeMethod.getParameterTypes(), invokeMethod.getGenericParameterTypes());
-                    RpcContext.getContext().setLocalAddress(channel.getLocalAddress()).setRemoteAddress(channel.getRemoteAddress());
+                    Object[] array = realize(list.toArray(), invokeMethod.getParameterTypes(),
+                            invokeMethod.getGenericParameterTypes());
+                    getContext().setLocalAddress(channel.getLocalAddress()).setRemoteAddress(channel.getRemoteAddress());
                     long start = System.currentTimeMillis();
-                    Object result = invoker.invoke(new RpcInvocation(invokeMethod, array)).recreate();
+                    RpcResult result = new RpcResult();
+                    try {
+                        Object o = invokeMethod.invoke(selectedProvider.getServiceInstance(), array);
+                        result.setValue(o);
```

Comment:
I doubt that invoking the underlying service instance directly bypassing Dubbo's proxy will make this `invoke` command meaningless. For a reading service, the result may be qualified as a reflection of that it will behave in production; for a writing service, the side effects may be not predictable

Reply:
@chickenlj would be possible to explain this in a bit more details (you might have already explained it, but for the shake of mine understanding it would great if you can again)

Reply:
@carryxyh . Got it. Thanks for expplaing it to me. What you saying is make sense and I think we allow to use the actual call path instead of by pass.

Reply:
@carryxyh I have created an issue [3059](https://github.com/apache/incubator-dubbo/issues/3059) for the same which we can use as a reffrence to work on.